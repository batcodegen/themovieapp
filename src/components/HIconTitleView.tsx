/*
 Horizontal view with title and icon
*/
import {StyleSheet, Text, View} from 'react-native';
import React, {ComponentProps} from 'react';
import Icon from 'react-native-vector-icons/AntDesign';
import {useTheme} from '@react-navigation/native';

interface HIconTitleViewProps {
  iconColor?: HEX | string;
  title: String;
  iconName: ComponentProps<typeof Icon>['name'];
}

const HIconTitleView = ({iconColor, title, iconName}: HIconTitleViewProps) => {
  const {colors} = useTheme();
  return (
    <View style={styles.container}>
      <Icon name={iconName} size={16} color={iconColor ?? colors.text} />
      <Text style={[styles.title, {color: colors.text}]}>{title}</Text>
    </View>
  );
};

export default HIconTitleView;

const styles = StyleSheet.create({
  container: {flexDirection: 'row', marginTop: 2},
  title: {marginStart: 5, textAlign: 'left'},
});
