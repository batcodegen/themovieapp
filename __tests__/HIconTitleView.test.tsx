import React from 'react';
import {render, screen} from '@testing-library/react-native';
import HIconTitleView from '../src/components/HIconTitleView';

describe('HIconTitleView Component', () => {
  it('Renders correctly', () => {
    render(<HIconTitleView iconName="menu" title={'title'} iconColor="#fff" />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
