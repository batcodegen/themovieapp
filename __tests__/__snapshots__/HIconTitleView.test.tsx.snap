// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HIconTitleView Component Renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "marginTop": 2,
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#fff",
          "fontSize": 16,
        },
        undefined,
        {
          "fontFamily": "anticon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    ?
  </Text>
  <Text
    style={
      [
        {
          "marginStart": 5,
          "textAlign": "left",
        },
        {
          "color": undefined,
        },
      ]
    }
  >
    title
  </Text>
</View>
`;
